### ISSUE TITLE
Proton 6.3-6 RC Testing

### ISSUE THREAD

**Issue by aeikum** [ValveSoftware/Proton#5065]
Hello. We've put together a new Proton 6.3 build for you to test.

👇
**But Wait! Please Read This First!** In this build, we have reconfigured where games store their save files in order to fix the Steam cloud sync feature for a number of games. Your save data will be migrated automatically and should continue to function even when switching between Proton versions. While we have done a good amount of testing on this and don't expect there to be any problems, there is **a possibility of losing your save data** when using this build. If you have very precious save data, we suggest you back up your data, or avoid using this build for now. On the flip side, we strongly appreciate any extra testing you can do with this build to help ensure there aren't any lurking save data issues. **Thank you for reading this paragraph.**
👆

In the Steam client, the Proton 6.3 app should have a "**proton-6.3-6-rc**" beta branch which you can choose to start testing the 6.3-6 Release Candidates. You may need to restart the Steam client to see it. Note that the name of the build in the Steam Settings dialog will not be updated. I will post changes here when we push new builds. The source for the latest RC build is available on the proton_6.3-next branch in these repositories. This branch may receive forced updates.

In this issue, we are interested only in problems that are new to the 6.3-6 RC builds. If you find new problems in the "next" branch, please confirm that the problem does not occur in on the "default" branch before reporting it here. **In particular, if you find your save data has changed in an unexpected way, please report that here.**

Here is the tentative changelog. As always, this changelog has not yet been verified by our QA staff, and can change before the final release as we add or remove features during RC testing.
- The following games are now playable:
  - Tokyo Xanadu eX+
  - Sonic Adventure 2
  - Rez Infinite
  - Elite Dangerous
  - Blood of Steel
  - Homeworld Remastered Collection
  - Star Wars Knights of the Old Republic
  - Guardians VR
  - 3D Aim Trainer

- Fix Microsoft Flight Simulator hang while loading.

- Fix Unreal Engine 4 update installation, which impacted Everspace 2 and KARDS.

- Better cursor capture behavior for fullscreen windows.

- Improve input handling after focus loss in some games like Warhammer: Chaosbane and Far Cry Primal.

- Correct game save locations to improve Steam cloud sync behavior in:
  - Guilty Gear -Strive-
  - Death Stranding
  - Katamari Damacy Reroll
  - Scarlet Nexus

- Fix text input and clipboard pasting in Origin overlay and launcher.

- Fix some music not playing in Planet Coaster.

- Improve non-English experience in Cyberpunk 2077 and Rockstar Games launchers.

- Improve launcher behavior in Swords of Legends Online.

- Fix Mafia III: Definitive Edition performance when FPS limiter is enabled.

- Fix audio playback issues in Fallout: New Vegas, Oblivion, Borderlands 3, and Deep Rock Galactic.

- Improve video playback in Deep Rock Galactic, The Medium, Nier: Replicant, and Contra: Rogue Corps.

- Update wine-mono to [6.3.0](https://github.com/madewokherd/wine-mono/releases/tag/wine-mono-6.3.0).

- Update DXVK to [v1.9.1](https://github.com/doitsujin/dxvk/releases/tag/v1.9.1).

- Update vkd3d-proton to [v2.4](https://github.com/HansKristian-Work/vkd3d-proton/releases/tag/v2.4).

- Update FAudio to [20.08](https://github.com/FNA-XNA/FAudio/releases/tag/21.08).

- Optionally support Nvidia's NVAPI GPU support library and DLSS. It is disabled by default and can be enabled with PROTON_ENABLE_NVAPI=1.

**Comment by AniLeo** [comment]
Fixes a performance regression present in Proton 6.3-5 with STEINS;GATE ([412830](https://store.steampowered.com/app/412830/STEINSGATE/)) that made the game run with poor performance and thus also run in slow motion.

This is fixed both in `proton-6.3-6-rc` and `Proton Experimental`.

Tested on:
CPU: Ryzen 7 2700X
GPU: Radeon RX 6800 XT
OS: Manjaro Linux 21.1.0
Kernel: Linux 5.13-9
Mesa: 21.1.6

**Comment by adolfintel** [comment]
Can someone point me to the commit that made Star Wars Knights of the Old Republic playable?
As far as I know, the game has frequent crashes on wine during area transitions regardless of hardware and software configuration, which makes it very annoying especially after Taris (just reload a saved game a dozen times: it will either crash or display corrupted graphics and then crash)

**Comment by aeikum** [comment]
@adolfintel It was a fix to the fullscreen hack. I've put notes in the git log to show what commits fix which games. It probably won't fix the crashes you're talking about, this was a fix for the game failing to render entirely.

**Comment by lanceturbes** [comment]
Confirming that Death Stranding doesn't have the cloud-save issue anymore, which is great! 

(EDIT: Removed irrelevant bug listed here caused by buggy NVIDIA driver, _not_ Proton.)

Tested on:
CPU: Ryzen 5 1600
GPU: GTX 1070 (NVIDIA Driver 460.91.03)
OS: Pop!_OS 21.04
Kernel: 5.11.0-7620-generic

**Comment by Leopard1907** [comment]
@WaywardPooch There is a problem on Nvidia 470 series with DX12 games right now that fills entire system ram pretty quickly ( due to that also causes high amount of swap usage too, which will make everything crawl and possibly act like it hanged) , such as noted below:


https://forums.developer.nvidia.com/t/high-ram-consumption-when-playing-dx12-games-nvidia-470-57-02-drivers/184635

So your problem might be related to that instead of Proton itself. You can observe ram usage with Mangohud etc while in game.

**Comment by lanceturbes** [comment]
@Leopard1907 You were right; downgraded my NVIDIA drivers to 460.91.03 and I don't get that problem anymore. Thanks for the suggestion!

**Comment by junglerobba** [comment]
Shin Megami Tensei 3 HD is another game where the cloud save locations are now fixed with this release, previously had to create some links to make those work.

**Comment by aeikum** [comment]
I've pushed an update to the 6.3-6 RC which should fix the Unreal Engine 4 installation error, which was not actually fixed in earlier builds.

**Comment by aeikum** [comment]
I've pushed an update to the 6.3-6 RC which removes the fix for Project Cars 3 focus loss because it caused a regression in Deus Ex GOTY.

**Comment by jr1991-g** [comment]
eFootball PES 2021 crashes at launch 


> I reinstall and test on Proton 6.3 RC, this time at least settings.exe can launch, but the game itself launches few seconds then crash.
> 
> [steam-1259970.log](https://github.com/ValveSoftware/Proton/files/6980267/steam-1259970.log)
> 
> I am able to launch settings and the game with Proton 5.13 after replacing wine mono with version 6.3.0, but can't download data assets.

https://github.com/ValveSoftware/Proton/issues/4199#issuecomment-898216437

**Comment by kallestrop** [comment]
After upgrading from Proton experimental-6.3-20210730 to Proton-6.3-6-rc4 Euro Truck Simulator 2 (227300) don't start anymore. 

I noticed several folder name changes in `~/.local/share/Steam/steamapps/compatdata/227300/pfx/drive_c/users/steamuser`

e.g. `My Documents` is now a symlink to `Documents`

I attached both tree outputs of the prefix folder. There are also some error messages in terminal output of Steam Client


[steam-227300.log](https://github.com/ValveSoftware/Proton/files/7011202/steam-227300.log)

[proton6.3-6rc4_terminal_output.log](https://github.com/ValveSoftware/Proton/files/7011201/proton6.3-6rc4_terminal_output.log)

[tree_steamusers_proton_6.3.5.txt](https://github.com/ValveSoftware/Proton/files/7011203/tree_steamusers_proton_6.3.5.txt)

[tree_steamusers_proton_6.3.6-rc4.txt](https://github.com/ValveSoftware/Proton/files/7011205/tree_steamusers_proton_6.3.6-rc4.txt)

**Comment by iWeaker** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/5065#issuecomment-901550805

Try to delete the game Prefix, and try again.

**Comment by kallestrop** [comment]
> 
> Try to delete the game Prefix, and try again.

After deleting prefix game is running again. Thanks!

**Comment by aeikum** [comment]
@kallestrop Thanks for reporting. Have you ever used an unofficial Proton build for that game, like a GE build or something? Trying to figure out what causes that problem.

**Comment by kallestrop** [comment]
@aeikum yes, i used Proton GE in the past for some benchmark tests. 

I can reproduce breaking the prefix when using Proton-6.14-GE-2:

1. Delete prefix and start game with Proton-6.3-6-rc4 --> game starts
2. Switch to Proton-6.14-GE-2 --> game doesn't start
3. Switch back to Proton-6.3-6-rc4 --> game doesn't start

I made the same test with latest Proton-6.15-GE-2:

1. Delete prefix and start game with Proton-6.3-6-rc4 --> game starts
2. Switch to Proton-6.15-GE-2 --> game starts
3. Switch back to Proton-6.3-6-rc4 --> game starts

So it seems that only using Proton-6.14-GE-2 breaks the Prefix.

**Comment by aeikum** [comment]
@kallestrop Thanks for the explanation. We should have a fix for this problem soon.

**Comment by alasky17** [comment]
@kallestrop and anyone else who tested the RC and also uses Proton GE builds  (and pinging @GloriousEggroll )  

TLDR: If you see a Steam cloud save conflict warning, and there is no timestamp associated with either the local or the cloud save option, **choose the cloud save option**.  

The final Proton 6.3-6b build (which should be live shortly), will have a fix for folks upgrading the prefix from Proton-6.14-GE-2.  While I was testing the build, I found that in some cases Steam would warn about a save conflict and ask me to choose between a cloud save or a local save, and neither of the saves would have a timestamp in the Steam dialogue box.  If you run into this warning, the **cloud save is the correct choice**.  Choosing the local save could result in losing your cloud save data and replacing it with nothing. In my testing, this shouldn't happen to folks who upgrade directly from 6.14-GE-2 to 6.3-6b directly (without testing any of the 6.3-6 rc builds).
